File: frontend/src/interfaces/reports/ReporteCargaDocenteResponse.ts
import { ReporteDocenteItem } from "./ReporteDocenteItem";

export interface ReporteCargaDocenteResponse {
    semestre: string;
    totalDocentes: number;
    docentesConCargaCompleta: number;
    docentesConCargaParcial: number;
    docentesConCargaExcedida: number;
    docentesSinAsignaciones: number;
    docentes: ReporteDocenteItem[];
}

File: backend/src/ApplicationSchedule.Application/DTOs/Horarios/ReporteCargaDocenteResponse.cs
namespace ApplicationSchedule.Application.DTOs.Horarios;

/// <summary>
/// DTO que representa el reporte de carga docente consolidado.
/// Incluye resumenes por docente y totales por escenario/periodo.
/// </summary>
public class ReporteCargaDocenteResponse
{
    public string Semestre { get; set; } = string.Empty;

    public int TotalDocentes { get; set; }

    public int DocentesConCargaCompleta { get; set; }

    public int DocentesConCargaParcial { get; set; }

    public int DocentesConCargaExcedida { get; set; }

    public int DocentesSinAsignaciones { get; set; }

    public List<ReporteDocenteItem> Docentes { get; set; } = new();
}